apps/dashboard/app/apps/nav_bar.rb

```diff
@@ -97,12 +97,13 @@ def self.nav_page(item, category, subcategory)
   def self.item_from_token(token)
     static_link_template = STATIC_LINKS.fetch(token.to_sym, nil)
     if static_link_template
-      return NavItemDecorator.new({}, static_link_template)
+      return NavItemDecorator.new({}, static_link_template, nil)
     end
 
     matched_apps = Router.pinned_apps_from_token(token, SysRouter.apps)
     if matched_apps.size == 1
-      extend_link(matched_apps.first.links.first) if matched_apps.first.links.first
+      app = AppRecategorizer.new(matched_apps.first, category: '', subcategory: '')
+      extend_link(matched_apps.first.links.first, apps: [app]) if matched_apps.first.links.first
```

Comment:
Is this why you're extending the initializers interface? Does this do the same thing?

```suggestion
      app = AppRecategorizer.new(matched_apps.first, category: '', subcategory: '')
      extend_link(app.links.first) if matched_apps.first.links.first
```

Reply:
will update....

Reply:
Sorry it seems like you had it the first time, I was trying to understand why you need `apps` and it would seem you do need to pass `[apps]` here.